pkg/controllers/plugin/util.go

```diff
@@ -53,3 +62,127 @@ func convertRuntimeObjectToCoreV1Service(o interface{}) (*corev1.Service, error)
 		return nil, fmt.Errorf("unsupported runtime.Object type: %T", obj)
 	}
 }
+
+func isPayloadReadyRunning(o interface{}) bool {
+	switch obj := o.(type) {
+	case *appsv1.Deployment:
+		if obj.Status.ReadyReplicas == obj.Status.Replicas && obj.Status.ReadyReplicas == obj.Status.AvailableReplicas {
+			return true
+		}
+	case *appsv1.StatefulSet:
+		if obj.Status.ReadyReplicas == obj.Status.Replicas && obj.Status.ReadyReplicas == obj.Status.AvailableReplicas {
+			return true
+		}
+	case *appsv1.DaemonSet:
+		if obj.Status.NumberReady == obj.Status.DesiredNumberScheduled && obj.Status.NumberReady == obj.Status.NumberAvailable {
+			return true
+		}
+	case *appsv1.ReplicaSet:
+		if obj.Status.ReadyReplicas == obj.Status.Replicas && obj.Status.ReadyReplicas == obj.Status.AvailableReplicas {
+			return true
+		}
+	case *batchv1.CronJob:
+		// CronJob does not have a status field just for the job, so we need to check the last successful time
+		if obj.Status.LastSuccessfulTime != nil {
+			return true
+		}
+	case *corev1.PodList:
+		// Check if all pods are running, if one of them is not running, return false
+		for _, pod := range obj.Items {
+			if pod.Status.Phase != corev1.PodRunning {
+				return false
+			}
+		}
+		return true
+	}
+	return false
+}
+
+func fetchPodList(labelSelector map[string]string, nameSpace string, cl client.Client) *[]PodStatus {
```

Comment:
NIT: namespace is one word, no need for camelcase here.
```suggestion
func fetchPodList(labelSelector map[string]string, namespace string, cl client.Client) *[]PodStatus {
```